### PR TITLE
Add Reducers and actions

### PR DESCRIPTION
**Pull request by StevenWafeek**
In this project we : 
- Add Redux Toolkit 
- Create a directory that will contain all your Redux logic
- Define a slice of state for books 
- Define a slice of state for categories

### CODE REVIEW

File: src/redux/store.js
Comment (ShadyShawkat) [inline src/redux/store.js:1]: - [ ] As per the requirements, the store should be configured using the redux library. You can do so by following this example:

```
const store = configureStore({
  reducer: { ...imported reducers },
})

```

File: src/redux/categories/categoriesSlice.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+const initialState = {
+  categories: [],
+  status: 'Under construction',
+};
+
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {
+  },
```
Comment (ShadyShawkat) [inline src/redux/categories/categoriesSlice.js]: - [ ] Kindly make sure to add a reducer here that checks the status and always returns "Under Construction", as per [this](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/add_reducers_an_actions_v2.md#:~:text=Includes%20a%20reducer%20that%20checks%20the%20status%20and%20always%20returns%20%22Under%20construction%22%20(the%20initial%20state%20should%20check%20to%20that%20string)) requirement.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @StevenWafeek, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AshakaE — APPROVED** [review]
Hi @StevenWafeek   , 

### Highlights ✨ 

- Thanks for making the requested changes 👍 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Proceed to the next level 👏 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**